### PR TITLE
Fixes #1 Fixes #2 Fixes #3

### LINKED ISSUE #1: Allow comparison tolerance for numeric arrays
Allow comparison tolerance for `double[]` and` float[]`

### LINKED ISSUE #2: Add a method to check that a string is null or empty
right now, one has to do 

```
 string.IsNullOrEmpty(_unknownValueOrigin.Description).ShouldBeTrue();
```

### LINKED ISSUE #3: Incorporate Async Context from PKSim

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/OSPSuite.BDDHelper/ContextSpecificationAsync.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+﻿using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace OSPSuite.BDDHelper
+{
+   /// <summary>
+   ///    Defines a static context specification used for static class or extensions methods
+   /// </summary>
+   [TestFixture]
+   public abstract class ContextSpecificationAsync<T>
```
Comment (msevestre) [inline src/OSPSuite.BDDHelper/ContextSpecificationAsync.cs:10]: This was defined in PKSim and needs to be move there now

File: src/OSPSuite.BDDHelper/Extensions/BDDExtensions.cs [hunk 2]
```diff
@@ -137,67 +146,60 @@ public static void ShouldOnlyContainInOrder<T>(this IEnumerable<T> items, IEnume
...
       }
 
       /// <summary>
-      ///    Asserts that the double <paramref name="actual" /> is equal to the double <paramref name="expected" /> within th
+      ///    Asserts that the double array <paramref name="actual" /> is equal to the double array <paramref name="expected" />
+      ///    within the
       ///    given relative tolerance.
       /// </summary>
-      public static void ShouldBeEqualTo(this double actual, double expected, double relTol)
+      public static void ShouldBeEqualTo(this double[] actual, double[] expected, double relTol, string message = null)
```
Comment (msevestre) [inline src/OSPSuite.BDDHelper/Extensions/BDDExtensions.cs:189]: added comparison of double arrays with tolerance

File: src/OSPSuite.BDDHelper/Extensions/BDDExtensions.cs [hunk 3]
```diff
@@ -229,15 +236,19 @@ public static void ShouldBeEqualTo(this float actual, float expected, double rel
       }
 
       /// <summary>
-      ///    Asserts that <paramref name="actual" /> is equal to <paramref name="expected" /> using the object.Equals comparer.
-      ///    Log the <paramref name="message" /> if not.
+      ///    Asserts that the string <paramref name="actual" /> is the null or empty string
       /// </summary>
-      public static void ShouldBeEqualTo<T>(this T actual, T expected, string message)
+      public static void ShouldBeNullOrEmpty(this string actual, string message = null)
```
Comment (msevestre) [inline src/OSPSuite.BDDHelper/Extensions/BDDExtensions.cs:241]: Helper function that I have been missing for  a while

File: src/OSPSuite.BDDHelper/Extensions/BDDExtensions.cs [hunk 4]
```diff
@@ -304,6 +306,25 @@ public static void ShouldNotBeEqualTo<T>(this T actual, T expected, string messa
          resultingException.ShouldBeAnInstanceOf<TException>();
       }
 
+      /// <summary>
+      ///    Asserts that <paramref name="workToPerform" /> throws an exception when performed.
+      /// </summary>
+      public static void ShouldThrowAn<TException>(this Func<Task> workToPerform) where TException : Exception
```
Comment (msevestre) [inline src/OSPSuite.BDDHelper/Extensions/BDDExtensions.cs:312]: Also new from PKSim

**Review by Yuri05 — COMMENTED** [review]
looks good

**Pull request merged**